crates/primitives/src/header.rs

```diff
@@ -117,6 +126,44 @@ impl Default for Header {
 }
 
 impl Header {
+    /// Performs a sanity check on the extradata field of the header.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the extradata size is larger than 100 KB.
+    pub fn check_extradata(&self) -> Result<(), HeaderError> {
+        if self.extra_data.len() > 100 * 1024 {
+            return Err(HeaderError::LargeExtraData);
+        }
+        Ok(())
+    }
+
+    /// Performs a sanity check on the block difficulty field of the header.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the block difficulty exceeds 80 bits.
+    pub fn check_difficulty(&self) -> Result<(), HeaderError> {
```

Comment:
naming nit:

these should be called `ensure_`

Reply:
done